src/Raven.Server/Documents/Patch/ScriptRunner.cs

```diff
@@ -1117,6 +1119,28 @@ public JsValue PutDocument(JsValue self, JsValue[] args)
                 }
             }
 
+            private string GenerateIdForShard(JsValue[] args, string id)
+            {
+                if (id?[^1] != _database.IdentityPartsSeparator)
+                    return id;
+
+                var originalId = (args[1].AsObject() as BlittableObjectInstance)?.DocumentId;
+                if (originalId != null)
+                {
+                    var builder = new StringBuilder(id);
+                    var index = originalId.IndexOf('$');
+                    if (index != -1)
+                        originalId = originalId[index..originalId.Length];
+                    else
+                        builder.Append('$');
+
+                    builder.Append(originalId + '$' + _database.IdentityPartsSeparator);
```

Comment:
I think we can use the `ExtractStickyId` method here, no?

Reply:
Also, we are already using string builder, let's avoid string concat

Reply:
> I think we can use the `ExtractStickyId` method here, no?

not sure I follow?.. :-)